chia/wallet/wallet_state_manager.py

```diff
@@ -1430,6 +1431,25 @@ async def get_all_transactions(self, wallet_id: int) -> List[TransactionRecord]:
     async def get_transaction(self, tx_id: bytes32) -> Optional[TransactionRecord]:
         return await self.tx_store.get_transaction_record(tx_id)
 
+    async def get_transaction_by_wallet_record(self, wr: WalletCoinRecord) -> Optional[TransactionRecord]:
+        records = await self.tx_store.get_transactions_by_height(wr.confirmed_block_height)
+        for record in records:
+            if wr.coin in record.additions:
+                return record
+        return None
+
+    async def get_coin_record_by_wallet_record(self, wr: WalletCoinRecord) -> Optional[CoinRecord]:
+        tx_record: Optional[TransactionRecord] = await self.get_transaction_by_wallet_record(wr)
+        if tx_record is None:
+            return None
+        return wr.to_coin_record(tx_record.created_at_time)
```

Comment:
Doing this is misleading.  The transaction `created_at_time` field is usually set with a `time.time()` and has nothing to do with the blockchain.  In most of the full node RPCs that return coin records, you'll see the timestamp of that coin's confirmation on chain, so this will not match what the actual coin record is.